python/lsst/ts/tunablelaser/read_serial_temp_scanner.py

```diff
@@ -0,0 +1,336 @@
+        logger object
+    port : `int`, optional
+        port that the server will be hosted on, default 1883
+    host : `str`, optional
+        IP the server will be hosted on, default tcpip.DEFAULT_LOCALHOST
+    encoding : `str`, optional
+        Encoding used for the packets
+    terminator: `bytes`, optional
+        terminating character used for packets
+    sample_wait_time: `int`, optional
+        time to wait between getting temperature samples
+    serial: `serial` or `None`, optional
+        serial object that the temperature scanner device is connected
+    temperature_windows: `int`, optional
+        Amount of temperature windows to average for rolling avg window
+
+    """
+
+    def __init__(
+        self,
+        logger: logging.Logger,
+        port: int | None = 1884,
+        host: str | None = tcpip.DEFAULT_LOCALHOST,
+        encoding: str = tcpip.DEFAULT_ENCODING,
+        terminator: bytes = tcpip.DEFAULT_TERMINATOR,
+        sample_wait_time: int = 5,
+        serial=None,
+        temperature_windows: int = 8,
+    ):
+        super().__init__(
+            log=logger,
+            port=port,
+            host=host,
+            connect_callback=None,
+            monitor_connection_interval=0,
+            name="",
+            encoding=encoding,
+            terminator=terminator,
+        )
+
+        self.serial = serial
+        self.sensor_dict = {}
+        self.fileptr = None
+        if logger is None:
+            logging.basicConfig(
+                format="%(asctime)s %(levelname)-8s %(message)s",
+                level=logging.INFO,
+                datefmt="%Y-%m-%d %H:%M:%S",
+            )
+
+            logger = logging.getLogger(__name__)
+            logger.propagate = True
+            self.logger = logger
+        else:
+            self.logger = logger
+        self.sample_wait_time = sample_wait_time
+
+        # Fan sensor
+        self.fan_sensor = ""
+        self.fan_gpio = None
+        self.fan_turn_on_temp = 0
+        self.fan_turn_off_temp = 0
+
+        self.latest_data = {sensor_name: 0 for sensor_name in self.sensor_dict}
+        self.rolling_temperature = [0 for _ in range(temperature_windows)]
+
+        self.pi = pigpio.pi()
+
+        self.configured = False
+        self.first_run = True
+        self.encoding = encoding
+        self.pending_messages = []
+        self.port = port
+        self.host = host
+
+        # TODO remove this debug stuff
+        self.header = None
+
+        self.config()
+
+    async def amain(self):
+        self.serial_temperature_task()
+
+    def publish_msg(self, msg):
+        self.pending_messages.append(msg)
+        self.read_and_dispatch()
+
+    async def read_and_dispatch(self) -> None:
+        # read any data
+        # incoming_data = self.read_str()
+
+        # deal with data
+
+        # send data
+        while len(self.pending_messages) != 0:
+            self.write_str(self.pending_messages.pop() + self.terminator)
```

Comment:
I think you are missing the await here.

Also, why are you doing it like this? why not just call `self.write_str` directly when you want to send a message?